src/submit.py

```diff
@@ -2,78 +2,103 @@
 
-_SUBMISSION_PATH = AGENT_HOME_DIR / "submission.txt"
+_SUBMISSION_PATH = settings.AGENT_HOME_DIR / "submission.txt"
 
 
-async def _git_push(repo_dir: pathlib.Path) -> tuple[int, str]:
+async def run_command(command: list[str], cwd: pathlib.Path) -> tuple[int, str]:
     process = await asyncio.subprocess.create_subprocess_exec(
-        "git",
-        "push",
-        cwd=repo_dir,
+        *command,
+        cwd=cwd,
         stdout=asyncio.subprocess.PIPE,
         stderr=asyncio.subprocess.STDOUT,
     )
     stdout, _ = await process.communicate()
-    output = stdout.decode().strip()
     return_code = await process.wait()
-    return return_code, output
+    return return_code, stdout.decode().strip()
+
+
+async def _git_push(repo_dir: pathlib.Path) -> tuple[int, str]:
+    return await run_command(["git", "push"], repo_dir)
 
 
 async def _create_submission_commit(repo_dir: pathlib.Path):
-    # Stash any changes
-    stash_process = await asyncio.subprocess.create_subprocess_exec(
-        "git",
-        "stash",
-        "push",
-        cwd=repo_dir,
-        stdout=asyncio.subprocess.PIPE,
-        stderr=asyncio.subprocess.STDOUT,
+    commands = [
+        ["git", "stash", "push"],
+        ["git", "commit", "--allow-empty", "-m", "SUBMISSION"],
+        ["git", "stash", "pop"],
+    ]
+    for command in commands:
+        await run_command(command, repo_dir)
+
+
+async def git_clone_instructions(repo_dir: pathlib.Path):
+    _, ip_address = await run_command(["hostname", "-I"], cwd=repo_dir)
+    get_origin_failed, origin_url = await run_command(
+        ["git", "remote", "get-url", "origin"], cwd=repo_dir
     )
-    await stash_process.communicate()
-
-    # Add empty commit
-    commit_process = await asyncio.subprocess.create_subprocess_exec(
-        "git",
-        "commit",
-        "--allow-empty",
-        "-m",
-        "SUBMISSION",
-        cwd=repo_dir,
-        stdout=asyncio.subprocess.PIPE,
-        stderr=asyncio.subprocess.STDOUT,
-    )
-    await commit_process.communicate()
-
-    # Pop stashed changes
-    pop_process = await asyncio.subprocess.create_subprocess_exec(
-        "git",
-        "stash",
-        "pop",
-        cwd=repo_dir,
-        stdout=asyncio.subprocess.PIPE,
-        stderr=asyncio.subprocess.STDOUT,
+    origin_var = ""
+    origin_desc = ""
+    if get_origin_failed or "No such remote" in origin_url:
+        origin_url = "$ORIGIN_URL"
+        origin_var = (
+            "ORIGIN_URL=git@github.com:evals-sandbox/baseline-TASK-YYYY-MM-DD-NAME.git"
+        )
+        origin_desc = (
+            " and replace TASK-YYYY-MM-DD-NAME with whatever\n"
+            "is in the name of the slack channel for this task"
+        )
+
+    _, jumphost = await run_command(["ssh", "-G", "metr-baseline"], cwd=repo_dir)
+    if match := re.search(r"proxyjump\s+(\w+)", jumphost):
+        jumphost = (
+            f' -o ProxyCommand=\\"ssh -i $SSH_KEY -W %h:%p ssh-user@{match.group(1)}\\"'
```

Comment:
`-J ssh-user@{match.group(1)}` usually works, but this is more general, so went with it

Reply:
But this introduces a bunch of escaping logic that I expect someone to break or the user to flub

Reply:
Added instructions for both, with `-J` being the default and a fallback to the more complicated one
It looks like this (I was playing about with .git/config):
![image](https://github.com/user-attachments/assets/aab97c5d-99ad-4682-9218-36e40118712f)